src/middleware/builtin.ts

```diff
@@ -231,7 +231,11 @@ export function ignoreSelf(): Middleware<AnyMiddlewareArgs> {
       }
 
       // Its an Events API event that isn't of type message, but the user ID might match our own app. Filter these out.
-      if (botUserId !== undefined && args.event.user === botUserId) {
+      // However, some events still must be fired, because they can make sense.
+      const eventsWhichShouldNotBeFilteredOut = ['member_joined_channel', 'member_left_channel'];
+      const isEventShouldBeSkipped = !eventsWhichShouldNotBeFilteredOut.includes(args.event.type);
```

Comment:
I see your double negative here.

```suggestion
      const isEventShouldBeSkipped = eventsWhichShouldBeSkipped.includes(args.event.type);
```